src/core/Song.cpp

```diff
@@ -644,39 +640,31 @@ void Song::stop()
 	// To avoid race conditions with the processing threads
 	Engine::audioEngine()->requestChangeInModel();
 
-	TimeLineWidget * tl = getPlayPos().m_timeLine;
+	auto& timeline = getTimeline();
 	m_paused = false;
 	m_recording = true;
+	m_playing = false;
 
-	if( tl )
+	switch (timeline.stopBehaviour())
 	{
-		switch( tl->behaviourAtStop() )
-		{
-			case TimeLineWidget::BehaviourAtStopState::BackToZero:
-				getPlayPos().setTicks(0);
-				m_elapsedMilliSeconds[static_cast<std::size_t>(m_playMode)] = 0;
-				break;
+		case Timeline::StopBehaviour::BackToZero:
+			getPlayPos().setTicks(0);
+			m_elapsedMilliSeconds[static_cast<std::size_t>(m_playMode)] = 0;
+			break;
```

Comment:
Is this something we do elsewhere? I tend to avoid doing so unless necessary (which is generally when a case declares a local variable, in which case its scope needs to be limited so as not to include the following case). It uses additional vertical space and doesn't really improve clarity, since the code is already indented relative to the case label.

We require explicit bracing for conditional statements and loops to help prevent bugs if code is inserted between the statement and its body, e.g.
```cpp
if (foo.has_value())
	// foo->bar();
	foo->baz(); // This line will run unconditionally if the line above is uncommented
```
This isn't a concern for cases in switch statements, as they behave somewhat differently.

Reply:
I tend to prefer them, but that's fine if we don't want to do that. 👍